printer/util.js

```diff
@@ -0,0 +1,71 @@
+const snmp = require('net-snmp');
+
+/**
+ * inkLevel determines and returns the current ink level of a HP p2015dn printer
+ * @param {String} printer IP address of printer to query
+ * @returns {Promise} Promise with current ink level of the printer in % of capacity
+ */
+async function inkLevel(printer) {
+  const currentLevel = await getCurrentTonerLevel(printer);
+  const capacity = await getTonerCapacity(printer);
+
+  return (currentLevel / capacity) * 100;
+}
+
+/**
+ * getTonerCapacity makes snmp query against toner capacity OID and returns value
+ * @param {String} printer IP address of printer to query
+ * @returns {Promise} Promise with toner capacity in unknown units
+ */
+function getTonerCapacity(printer) {
+  return new Promise((resolve) => {
+    const session = snmp.createSession(printer, 'public');
+
+    const oids = ['1.3.6.1.2.1.43.11.1.1.8.1.1'];
+
+    session.get(oids, function (error, varbinds) {
```

Comment:
dry principle. we are replicating `session.get` twice, this could be its own function like

```js
async function executeSnmpRequest(printerIp, objectIdentifiers) {
  return new Promise((resolve) => {
    const session = snmp.createSession(printerIp, 'public');
    try {
      session.get(objectIdentifiers, function (error, result) {
        if (error) {
          resolve(false);
        } else {
          resolve(result[0].value);
        }
        session.close();
      });

      session.trap(snmp.TrapType.LinkDown, function (error) {
        if (error) {
          resolve(false);
        }
      });
    } catch (e) {
      resolve(false);
    }
  });
}
```

ps `oid` is short for "object identifier" ([source](https://www.dpstele.com/snmp/what-does-oid-network-elements.php)) so the parameter naming promotes readability.

the above function also wraps the snmp request in a try/catch. as a result any errors makes the function resolve `false`.


then each function can call on `executeSnmpRequest` like:

```js
async function getTonerCapacity(printerIp) {
  await executeSnmpRequest(printerIp, /* object identifiers */)
}
```